### ISSUE TITLE
JASP does not build in spite of seemingly satisfied dependencies

### ISSUE THREAD

**Issue by anbiro** [jasp-stats/jasp-issues#499]
At university, we are using JASP for statistical analyses, so I need to install it. I read the instructions on https://jasp-stats.org/download/ for the Linux installation, but the flatpak method did not work (I got the message socket I/O timed out), so I tried to build it from source. 

Dependencies seem to be satisfied, but during the build process I got the error message:
```
Makefile: 4091: recipe for target 'analysisform.o' failed
make: *** [analysisform.o] Error 1
```
After this, I read into the Makefile, and found some R libraries on the INCPATH that were not present on my PC, though the required R packages r-base-core, r-cran-rcpp and r-cran-rinside are installed, but the script installExpectedPackages.R in the Tools directory did not run correctly. The missing libraries are /usr/include/R, /usr/lib/R/library/include, /usr/lib/R/library/Rcpp/include and /usr/lib/R/library/include. I also read that in some situations I just need to create a symbolic link to the libraries and have the question whether I should do this.

**Comment by JorisGoosen** [comment]
Well, you might want to retry installing the flatpak version, because it seems that your connection simply failed. And the new version is up there.

If you are determined to build it I am more then willing to help you out but I would need more information:
-  What version of linux are you running and on which architecture
- Did you try running `installExpectedPackages.R` as root? Because it does need the permission to write into the system R-library.
- Maybe you can add the entire output of the build-proces zipped to this thread?

But it is probably easiest if you just try flatpak again (and make sure you are running the latest flatpak version).

**Comment by anbiro** [comment]
Thank you for support and sorry for being late with the answer.
I run flatpak more times and again after trying to build JASP, but I
always get the same error. The interaction is the following:
sudo flatpak install http://static.jasp-stats.org/flatpak/JASP.flatpakref
Installing: org.jaspstats.JASP/x86_64/master
Required runtime for org.jaspstats.JASP/x86_64/master
(org.kde.Sdk/x86_64/5.12) is not installed, searching...
Found in remote flathub, do you want to install it? [y/n]: y
Installing: org.kde.Sdk/x86_64/5.12 from flathub

Receiving delta parts: 7/22 722.2 kB/s 161.8 MB/740.1 MB 13 minutes 20
seconds remaining
error: While pulling runtime/org.kde.Sdk/x86_64/5.12 from remote
flathub: Socket I/O timed out

It is known that someone is harassing me and sometimes he breaks the
connection. The output of installExpectedPackages.R looks like this:
~/Downloads/jasp-desktop-development/Tools$ sudo Rscript
installExpectedPackages.R
[1] "Run createFlatpakJson() to transform the expected packages of
JASP into a fresh org.jaspstats.JASP.json for flatpak.\nOr run
installRequiredPackages() as administrator to get you local installed
version of R up to speed with the same packages."

The Linux version is Debian 9.3, architecture is amd64, and I hope
that I zipped the build process correctly (GitHub told me that
something went really wrong and it could not process the compressed
file).

2019-09-25 13:31 GMT+02:00, Joris Goosen <notifications@github.com>:
> Well, you might want to retry installing the flatpak version, because it
> seems that your connection simply failed. And the new version is up there.
>
> If you are determined to build it I am more then willing to help you out but
> I would need more information:
> -  What version of linux are you running and on which architecture
> - Did you try running `installExpectedPackages.R` as root? Because it does
> need the permission to write into the system R-library.
> - Maybe you can add the entire output of the build-proces zipped to this
> thread?
>
> But it is probably easiest if you just try flatpak again (and make sure you
> are running the latest flatpak version).
>
> --
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/jasp-stats/jasp-issues/issues/499#issuecomment-534978870


-- 
Krisztina

**Comment by JorisGoosen** [comment]
Hey @anbiro if your connection keeps being cutoff it will be very hard to install anything that requires downloading stuff.

Maybe it would be smart to make sure that whoever is harassing you doesn't break your connection? (And in fact has no control over your connection whatsoever...)

If your connection actually works then you can just install the flathub version of JASP.

**Comment by anbiro** [comment]
Dear Joris,

the description of the problematic JASP installation was on the side
https://jasp-stats.org/linux-installation-guide that gave the result
that I described earlier:

sudo flatpak install http://static.jasp-stats.org/flatpak/JASP.flatpakref
Installing: org.jaspstats.JASP/x86_64/master
Required runtime for org.jaspstats.JASP/x86_64/master
(org.kde.Sdk/x86_64/5.12) is not installed, searching...
Found in remote flathub, do you want to install it? [y/n]: y
Installing: org.kde.Sdk/x86_64/5.12 from flathub

Receiving delta parts: 7/22 722.2 kB/s 161.8 MB/740.1 MB 13 minutes 20
seconds remaining
error: While pulling runtime/org.kde.Sdk/x86_64/5.12 from remote
flathub: Socket I/O timed out

...
Later I read another instruction on another side,
https://flathub.org/apps/details/org.jaspstats.JASP , that worked correctly:

sudo flatpak remote-add –if not-exists flathub
https://flathub.org/repo/flathub.flatpakrepo
sudo shutdown -r now
[…]
sudo flatpak install flathub org.jaspstats.JASP

This solved my problem, but it was quite complicated (I was using
flatpak for the first time).

2019-11-06 12:59 GMT+01:00, Joris Goosen <notifications@github.com>:
> Closed #499.
>
> --
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/jasp-stats/jasp-issues/issues/499#event-2775035779


-- 
Krisztina

**Comment by JorisGoosen** [comment]
Aah yeah those were the old instructions. I just noticed that they were a bit unclear. I will make it a bit clearer so that the next person who is unfamiliar with flatpak will not run into the same problems as you did!
